OpenRA.Game/Server/PlayerMessageTracker.cs

```diff
@@ -0,0 +1,81 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenRA.Server
+{
+	class PlayerMessageTracker
+	{
+		readonly Dictionary<int, List<long>> messageTracker = new Dictionary<int, List<long>>();
+		readonly Server server;
+		readonly Action<Connection, int, int, byte[]> dispatchOrdersToClient;
+		readonly Action<Connection, string, Dictionary<string, object>> sendLocalizedMessageTo;
+
+		public PlayerMessageTracker(Server server, Action<Connection, int, int, byte[]> dispatchOrdersToClient, Action<Connection, string, Dictionary<string, object>> sendLocalizedMessageTo)
+		{
+			this.server = server;
+			this.dispatchOrdersToClient = dispatchOrdersToClient;
+			this.sendLocalizedMessageTo = sendLocalizedMessageTo;
+		}
+
+		public void DisableChat(Connection conn, int time)
+		{
+			dispatchOrdersToClient(conn, 0, 0, new Order("DisableChatEntry", null, false) { ExtraData = (uint)time }.Serialize());
+		}
+
+		public bool IsPlayerAtFloodLimit(Connection conn, string chatDisabled)
```

Comment:
The `ChatDisabled` translation reference is used only in this method so it can be moved to `PlayerMessageTracker` instead of passing it here from `Server`.

Reply:
Good catch. I've cashed `ChatDisabled` message to `PlayerMessageTracker`

Reply:
I meant to move as a static field of `PlayerMessageTracker` and remove it from `Server`.

Reply:
Ah, I get it. Moved